src/clientHandler.ts

```diff
@@ -12,298 +10,177 @@ import {
-      ignoreDirectoryNames = config('terraform-ls').get('ignoreDirectoryNames');
-      outputChannel = vscode.window.createOutputChannel(channelName);
-      outputChannel.appendLine(`Launching language server: ${cmd} ${serverArgs.join(' ')}`);
-    }
+    const initializationOptions = this.getInitializationOptions(commandPrefix);
 
-    if (rootModulePaths.length > 0 && excludeModulePaths.length > 0) {
-      throw new Error(
-        'Only one of rootModules and excludeRootModules can be set at the same time, please remove the conflicting config and reload',
-      );
-    }
+    const serverOptions: ServerOptions = this.getServerOptions();
+    this.outputChannel.appendLine(
+      `Launching language server: ${serverOptions.run.command} ${serverOptions.run.args.join(' ')}`,
+    );
 
-    const commandPrefix = this.shortUid.seq();
-    let initializationOptions = { commandPrefix, experimentalFeatures };
-    if (terraformExecPath.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { terraformExecPath });
-    }
-    if (terraformExecTimeout.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { terraformExecTimeout });
-    }
-    if (terraformLogFilePath.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { terraformLogFilePath });
-    }
-    if (rootModulePaths.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { rootModulePaths });
-    }
-    if (excludeModulePaths.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { excludeModulePaths });
-    }
-    if (ignoreDirectoryNames.length > 0) {
-      initializationOptions = Object.assign(initializationOptions, { ignoreDirectoryNames });
-    }
+    const documentSelector: DocumentSelector = [
+      { scheme: 'file', language: 'terraform' },
+      { scheme: 'file', language: 'terraform-vars' },
+    ];
 
-    const executable: Executable = {
-      command: cmd,
-      args: serverArgs,
-      options: {},
-    };
-    const serverOptions: ServerOptions = {
-      run: executable,
-      debug: executable,
-    };
     const clientOptions: LanguageClientOptions = {
       documentSelector: documentSelector,
-      workspaceFolder: wsFolder,
       initializationOptions: initializationOptions,
       initializationFailedHandler: (error) => {
         this.reporter.sendTelemetryException(error);
         return false;
       },
-      outputChannel: outputChannel,
+      outputChannel: this.outputChannel,
       revealOutputChannelOn: RevealOutputChannelOn.Never,
     };
 
-    const client = new LanguageClient(id, name, serverOptions, clientOptions);
+    const id = `terraform`;
+    const client = new LanguageClient(id, serverOptions, clientOptions);
 
+    const enableReferenceCountCodeLens: boolean = config('terraform').get('enableReferenceCountCodeLens');
```

Comment:
After getting rid of all the let variables, we can now pass the type to get and get rid of the cast. 

```suggestion
    const enableReferenceCountCodeLens = config('terraform').get<v>('enableReferenceCountCodeLens');
```